src/libraries/Microsoft.PowerFx.Interpreter/Functions/StandardErrorHandling.cs

```diff
@@ -528,6 +551,65 @@ private static FormulaValue TimeOrDateTime(IRContext irContext, int index, Formu
         #endregion
 
         #region Common Runtime Value Checking Pipeline Stages
+        private static ErrorValue FiniteArgumentCheck(string functionName, IRContext irContext, FormulaValue[] args)
+        {
+            List<ErrorValue> errors = null;
+            foreach (var arg in args)
+            {
+                if (arg is ErrorValue ev)
+                {
+                    if (errors == null)
+                    {
+                        errors = new List<ErrorValue>();
+                    }
+
+                    errors.Add(ev);
+                }
+
+                if (arg is NumberValue nv && IsInvalidDouble(nv.Value))
+                {
+                    if (errors == null)
+                    {
+                        errors = new List<ErrorValue>();
+                    }
+
+                    errors.Add(new ErrorValue(irContext, new ExpressionError()
+                    {
+                        Message = $"Arguments to the {functionName} function must be finite.",
```

Comment:
>Arguments [](http://example.com/codeflow?start=36&length=9)

Should the error message indicate which argument? I don't know if we have this information to put in error message here.

Reply:
While we could pass this information through (which argument), we should never have a non-finite value flowing into the functions (none of the functions can return it), so I don't think we need to have different error messages for different arguments.